tests/integration/test_insert_into_distributed/test.py

```diff
@@ -162,6 +174,15 @@ def test_inserts_local(started_cluster):
     assert instance.query("SELECT count(*) FROM local").strip() == '1'
 
 
+def test_inserts_single_replica(started_cluster):
+    node1.query(
+        "INSERT INTO distributed_one_replica VALUES ('2000-01-01', 1)",
+        settings={"insert_distributed_sync": "1", "prefer_localhost_replica": "0"},
+    )
+    time.sleep(0.5)
+    assert node2.query("SELECT count(*) FROM single_replicated").strip() == '1'
+
+
```

Comment:
Also I would check for `internal_replication=false` too (that it will fail)

Reply:
Although I was bitten by this, I stil don't think we need this test because `internal_replication=false` is rarely used.

Reply:
> internal_replication=false is rarely used.

But if this switch exists it should work correctly, and checking that it is not broken after this change does not looks complex, take a look at the last patch from [this topic branch](https://github.com/azat/ClickHouse/pull/new/pr/23152) - https://github.com/ClickHouse/ClickHouse/commit/3f04db2ec671c487853424844925e606e12771da

*It also contains some tunning of existing test to make it deterministic, since before it uses random node*

Reply:
During testing this I observed one minor issue, fixed in #23744, but this affects the exception message in tests, so I pushed one more patch (on top of the previous), but it requires referenced PR.

Reply:
@azat I reviewed https://github.com/ClickHouse/ClickHouse/pull/23744. Can this PR be merged ? Or we need to wait https://github.com/ClickHouse/ClickHouse/pull/23744 first ?

Reply:
This can be merged as-is, but I would improve test, like here https://github.com/ClickHouse/ClickHouse/commit/3f04db2ec671c487853424844925e606e12771da

Reply:
#23751